File: Serene/Serene.Script/Imports/ServiceContracts/Northwind.EmployeeTerritoryService.cs
namespace Serene.Northwind
{
    using jQueryApi;
    using Serenity;
    using System;
    using System.Collections;
    using System.Collections.Generic;
    using System.Runtime.CompilerServices;

    public partial class EmployeeTerritoryService
    {
        [InlineConstant] public const string BaseUrl = "Northwind/EmployeeTerritory";
    
        public static jQueryXmlHttpRequest Create(SaveRequest<EmployeeTerritoryRow> request, Action<SaveResponse> onSuccess, ServiceCallOptions options = null)
        {
            return Q.ServiceRequest(Methods.Create, request, onSuccess, options);
        }
    
        public static jQueryXmlHttpRequest Update(SaveRequest<EmployeeTerritoryRow> request, Action<SaveResponse> onSuccess, ServiceCallOptions options = null)
        {
            return Q.ServiceRequest(Methods.Update, request, onSuccess, options);
        }
    
        public static jQueryXmlHttpRequest Delete(DeleteRequest request, Action<DeleteResponse> onSuccess, ServiceCallOptions options = null)
        {
            return Q.ServiceRequest(Methods.Delete, request, onSuccess, options);
        }
    
        public static jQueryXmlHttpRequest Retrieve(RetrieveRequest request, Action<RetrieveResponse<EmployeeTerritoryRow>> onSuccess, ServiceCallOptions options = null)
        {
            return Q.ServiceRequest(Methods.Retrieve, request, onSuccess, options);
        }
    
        public static jQueryXmlHttpRequest List(ListRequest request, Action<ListResponse<EmployeeTerritoryRow>> onSuccess, ServiceCallOptions options = null)
        {
            return Q.ServiceRequest(Methods.List, request, onSuccess, options);
        }
    
        [Imported, PreserveMemberCase]
        public static class Methods
        {
            [InlineConstant] public const string Create = "Northwind/EmployeeTerritory/Create";
            [InlineConstant] public const string Update = "Northwind/EmployeeTerritory/Update";
            [InlineConstant] public const string Delete = "Northwind/EmployeeTerritory/Delete";
            [InlineConstant] public const string Retrieve = "Northwind/EmployeeTerritory/Retrieve";
            [InlineConstant] public const string List = "Northwind/EmployeeTerritory/List";
        }
    }
    
}

File: Serene/Serene.Web/Modules/Imports/ServerTypings/Northwind.EmployeeTerritoryService.ts
namespace Serene.Northwind {
    export namespace EmployeeTerritoryService {
        export const baseUrl = 'Northwind/EmployeeTerritory';

        export declare function Create(request: Serenity.SaveRequest<EmployeeTerritoryRow>, onSuccess?: (response: Serenity.SaveResponse) => void, opt?: Serenity.ServiceOptions<any>): JQueryXHR;
        export declare function Update(request: Serenity.SaveRequest<EmployeeTerritoryRow>, onSuccess?: (response: Serenity.SaveResponse) => void, opt?: Serenity.ServiceOptions<any>): JQueryXHR;
        export declare function Delete(request: Serenity.DeleteRequest, onSuccess?: (response: Serenity.DeleteResponse) => void, opt?: Serenity.ServiceOptions<any>): JQueryXHR;
        export declare function Retrieve(request: Serenity.RetrieveRequest, onSuccess?: (response: Serenity.RetrieveResponse<EmployeeTerritoryRow>) => void, opt?: Serenity.ServiceOptions<any>): JQueryXHR;
        export declare function List(request: Serenity.ListRequest, onSuccess?: (response: Serenity.ListResponse<EmployeeTerritoryRow>) => void, opt?: Serenity.ServiceOptions<any>): JQueryXHR;

        export namespace Methods {
            export declare const Create: 'Northwind/EmployeeTerritory/Create';
            export declare const Update: 'Northwind/EmployeeTerritory/Update';
            export declare const Delete: 'Northwind/EmployeeTerritory/Delete';
            export declare const Retrieve: 'Northwind/EmployeeTerritory/Retrieve';
            export declare const List: 'Northwind/EmployeeTerritory/List';
        }

        ['Create', 'Update', 'Delete', 'Retrieve', 'List'].forEach(x => {
            (<any>EmployeeTerritoryService)[x] = function (r, s, o) { return Q.serviceRequest(baseUrl + '/' + x, r, s, o); };
            (<any>Methods)[x] = baseUrl + '/' + x;
        });
    }
}

